### PR TITLE
feat: exchanges filters

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

Depends on https://github.com/ArdentHQ/arkscan/pull/229

Closes https://app.clickup.com/t/861mruat3

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my UI changes in light AND dark mode
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: app/Http/Livewire/ExchangeTable.php [hunk 1]
```diff
@@ -20,6 +37,20 @@ public function render(): View
...
+            ->filterByType($this->type)
+            ->filterByPair($this->pair)
+            ->get();
+    }
+
+    public function setFilter(string $param, string $value): void
+    {
+        if ($param === 'type') {
+            $this->type = $value;
+        }
```
Comment (alfonsobries) [inline app/Http/Livewire/ExchangeTable.php:50]: I can also use $this->{$param} = $type but that creates a rector cs-fixer warning, also is safer this way

Comment (alexbarnsley) [reply app/Http/Livewire/ExchangeTable.php:50]: You could also validate the param beforehand if you wanted to do the `->{$param}` (assuming you can fix the warning)

```suggestion
    public function setFilter(string $param, string $value): void
    {
        if (! in_array($param, ['type', 'pair'])) {
            return;
        }
        
        $this->{$param} = $type;
    }
```

Comment (alexbarnsley) [reply app/Http/Livewire/ExchangeTable.php:50]: What was the warning?

File: app/Http/Livewire/ExchangeTableFilter.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        return view('livewire.exchange-table-filter', [
+            'selectedType' => in_array($this->type, array_keys($this->types), true) ? $this->type : 'all',
+            'selectedPair' => in_array($this->pair, array_keys($this->pairs), true) ? $this->pair : 'all',
+        ]);
+    }
+
+    public function setFilter(string $param, string $value): void
+    {
+        if ($param === 'type') {
```
Comment (alexbarnsley) [inline app/Http/Livewire/ExchangeTableFilter.php:54]: Could do the same as above too

File: app/Models/Exchange.php [hunk 3]
```diff
@@ -4,10 +4,33 @@
...
+    {
+        if ($type === 'exchanges') {
+            return $query->where('is_exchange', true);
+        }
+
+        if ($type === 'aggregators') {
+            return $query->where('is_aggregator', true);
+        }
+
+        return $query;
```
Comment (alexbarnsley) [inline app/Models/Exchange.php:25]: Alternative approach if you're interested:

```suggestion
        return $query
            ->when($type === 'exchanges', static fn ($query) => $query->where('is_exchange', true))
            ->when($type === 'aggregators', static fn ($query) => $query->where('is_aggregator', true));
```

**Review by alexbarnsley — COMMENTED** [review]
I know it's draft but added a few comments. Nothing wrong, just some alternative approaches

File: app/Http/Livewire/ExchangeTable.php (hunk 1 above)
Comment (alfonsobries) [reply app/Http/Livewire/ExchangeTable.php:50]: @alexbarnsley `Variable property access on $this(App\Http\Livewire\ExchangeTable)`, tried your suggestion but does not handle the warning

File: resources/views/components/exchanges/filter-dropdown.blade.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+<x-exchanges.dropdown
+    :title="trans($options[$selected])"
+    :icon="$icon"
+>
+    @foreach ($options as $value => $label)
+        @if ($value === $selected)
+            @continue
+        @endif
+
```
Comment (alexbarnsley) [inline resources/views/components/exchanges/filter-dropdown.blade.php:11]: according to designs, we show and highlight the selected option :thinking: 

Design:
image

Imp:
image

**Review by alexbarnsley — CHANGES_REQUESTED** [review]
Seems to hide the selected option whereas design doesn't. The filtering itself works well :+1:

**Comment by alfonsobries** [comment]
@alexbarnsley fixed

**Review by alexbarnsley — APPROVED** [review]
Gorgeous

**Pull request merged**